app.rb

```diff
@@ -0,0 +1,77 @@
+require_relative './lib/items_repository'
+require_relative './lib/orders_repository'
+
+class Application
+
+  # The Application class initializer
+  # takes four arguments:
+  #  * The database name to call `DatabaseConnection.connect`
+  #  * the Kernel object as `io` (so we can mock the IO in our tests)
+  #  * the ItemRepository object (or a double of it)
+  #  * the OrdersRepository object (or a double of it)
+  def initialize(database_name, io, items_repository, orders_repository)
+    DatabaseConnection.connect(database_name)
+    @io = io
+    @items_repository = items_repository
+    @orders_repository = orders_repository
+  end
+
+  def run
```

Comment:
I would try to make this method shorter, by splitting each chunk into it's own specific method